coins/src/utils/coins3/produce.ts

```diff
@@ -18,99 +21,122 @@ export type Dynamo = {
 async function produceTopics(
   items: Dynamo[],
   topic: Topic,
-  producer: Producer,
+  producer: Producer
 ) {
   const messages: string[] = [];
 
   items.map((item) => {
     const { symbol, decimals, price, SK } = item;
-    if (topic != "coins-timeseries" && SK != 0) return;
-    if (topic != "coins-metadata" && !price) return;
-    if (topic == "coins-metadata" && (!symbol || decimals == null)) return;
-    const messageObject: object = convertToMessage(item, topic);
+    if (topic !== "coins-timeseries" && SK !== 0) return;
+    if (topic !== "coins-metadata" && !price) return;
+    if (topic === "coins-metadata" && (!symbol || decimals == null)) return;
+
+    const messageObject = convertToMessage(item, topic);
     validate(messageObject, topic);
+
     const message = JSON.stringify(messageObject);
-    if (messages.includes(message)) {
-      message;
-      return;
+    if (!messages.includes(message)) {
+      messages.push(message);
     }
-    messages.push(message);
   });
 
-  await producer.send({
-    topic: `${topic}`,
-    messages: messages.map((value) => ({ value })),
-  });
+  if (!messages.length) return;
+
+  if (process.env.KAFKA_UP === 'true') {
+    try {
+      await producer.send({
+        topic: `${topic}`,
+        messages: messages.map((value) => ({ value })),
+      });
+      return;
+    } catch (error) {
+      console.error(`Kafka failed for topic "${topic}". Fallback to ES/Redis.`, error);
+    }
+  }
+
+  for (const msg of messages) {
+    const parsed = JSON.parse(msg);
+
+    if (topic === "coins-current") {
+      const redisClient = getRedis();
+      const redisKey = `price_${parsed.pid}`;
+      await redisClient.set(redisKey, msg, "EX", 24 * 3600);
+    } else if (topic === "coins-metadata") {
+      await init();
+      const esRecord = getMetadataRecord(parsed);
+      if (!esRecord) continue;
+      await esClient.index({
```

Comment:
I think both, redis and ES support writing multiple items in one call, should switch to that. 


Also, need to wrap this part in try catch, else for example, if init or getRedis call fails because we didnt configure either, then this escalates and rest of the code also fail (the other part this writes to ddb stops now)